arbos/parse_l2.go

```diff
@@ -250,6 +268,24 @@ func parseEspressoMsg(rd io.Reader) ([]espressoTypes.Bytes, *arbostypes.Espresso
 			}
 			txs = append(txs, nextMsg)
 		}
+	case L2MessageKind_EspressoSovereignTx:
+		nextMsg, err := util.BytestringFromReader(rd, arbostypes.MaxL2MessageSize)
+		if err != nil {
+			return nil, nil, err
+		}
+		jst := new(arbostypes.EspressoBlockJustification)
+		s := []byte{}
+		if err := rlp.DecodeBytes(nextMsg, &s); err != nil {
+			return nil, nil, err
+		}
+		if err := json.Unmarshal(s, jst); err != nil {
```

Comment:
I know this is done as before but it still reads a bit strange to have JSON serialized data inside RLP encoded data, if that is really what it is so suggest adding some comments for the reader who doesn't have all the context.

Reply:
The difference between this and the EspressoTx is that EspressoTx supports more than one message? Sorry have a bit of a hard time reading our existing implementation.

Reply:
`EspressoTx` is the type that we fetches from all the transactions from a hotshot block. We use here.
https://github.com/EspressoSystems/nitro-espresso-integration/blob/ee9c8674a969758e57a187975b88ba56c54b8b69/arbos/parse_l2.go#L528

We build the own type for easily parsing and distinguishing it from other message types.

As for the `EspressoSovereignTx` message, though it has the similar format, which looks like  `[type, jst, bytes]`, is built in the different way. `bytes` in the `EspressoTx` message is created by the the transactions from a hotshot block. And in the `EspressoSovereignTx`, `bytes` is the original `l2 msg` created by the centralized sequencer. And according to the design, we should send the whole block(l2msg) to hotshot. It results that we would just get only 1 big hotshot transaction.

Reply:
Thanks for the explanation.